lib/src/app/app.dart

```diff
@@ -92,15 +271,178 @@ class AppClient {
   }
 
   /// Get a stream of [LogEntry] for a specific [RobotPart]. Logs are sorted by descending time (newest first)
-  Stream<List<LogEntry>> tailLogs(RobotPart part, {bool errorsOnly = false}) {
+  Stream<List<LogEntry>> tailLogs(RobotPart part, {bool errorsOnly = false, String? filter}) {
     final request = TailRobotPartLogsRequest()
       ..id = part.id
-      ..errorsOnly = errorsOnly;
+      ..errorsOnly = errorsOnly
+      ..filter = filter ?? '';
     final response = _client.tailRobotPartLogs(request);
     final stream = response.map((event) => event.logs);
     return stream.asBroadcastStream(onCancel: (_) => response.cancel());
   }
 
+  /// Get a specific [RobotPart] history by ID
+  Future<List<RobotPartHistoryEntry>> getRobotPartHistory(String id) async {
+    final request = GetRobotPartHistoryRequest()..id = id;
+    final GetRobotPartHistoryResponse response = await _client.getRobotPartHistory(request);
+    return response.history;
+  }
+
+  /// Update a specific [RobotPart] by ID
+  Future<RobotPart> updateRobotPart(String partId, String name, Struct robotConfig) async {
```

Comment:
It looks like the idiom elsewhere in the SDK is to ask people to pass `struct`-like objects as a `Map<String, dynamic>` and then convert to a struct under the hood. Can we change to doing so here (and elsewhere where we ask for a `Struct` param)?